test/http_test.c

```diff
@@ -79,11 +83,15 @@ static int mock_http_server(BIO *in, BIO *out, char version, int keep_alive,
                        version == '0' ? "keep-alive" : "close") <= 0)
             return 0;
     if (is_get) { /* construct new header and body */
-        if ((len = ASN1_item_i2d(rsp, NULL, it)) <= 0)
+        if (txt != NULL)
+            len = strlen(txt);
+        else if ((len = ASN1_item_i2d(rsp, NULL, it)) <= 0)
             return 0;
-        if (BIO_printf(out, "Content-Type: application/x-x509-ca-cert\r\n"
-                       "Content-Length: %d\r\n\r\n", len) <= 0)
+        if (BIO_printf(out, "Content-Type: %s\r\n"
+                       "Content-Length: %d\r\n\r\n", content_type, len) <= 0)
```

Comment:
when I compile this with gcc-4.8.x I get a warning here:
```
test/http_test.c: In function 'test_http_keep_alive':
test/http_test.c:211:5: error: initializer element is not computable at load time [-Werror]
     server_args mock_args = { NULL, content_type, NULL, '0', 0 };
     ^
cc1: all warnings being treated as errors
```

Reply:
gcc-4.8.x appears rather old.
The code section you highlighted above was a bit confusing, but given the compiler warning message I found what the issue is. Fix is in #17308.